audit.go

```diff
@@ -201,21 +211,42 @@ func main() {
 	//Main loop. Get data from netlink and send it to the json lib for processing
 	for {
 		msg, err := nlClient.Receive()
-		timing := metric.GetClient().NewTiming() // measure latency from recipt of message
 		if err != nil {
 			if err.Error() == "no buffer space available" {
 				metric.GetClient().Increment("messages.netlink_dropped")
 			}
 			slog.Error.Printf("Error during message receive: %+v\n", err)
 			continue
 		}
-
-		metric.GetClient().Increment("messages.total")
 		if msg == nil {
 			continue
 		}
+		// As soon as we have a message, spawn a goroutine to handle it and free up the main loop
+		go handleMsg(msg, marshaller)
+	}
+}
+
+// Fetch the max value we can set from /proc/sys/net/core/rmem_max
+// This value is mounted in from the host via the kube yaml
+func fetchRmemMax() int {
+	var rmemMax int
+	file, err := os.Open("/proc/sys/net/core/rmem_max")
```

Comment:
Will this value reliably be present in the same place on all linux distros?  Wondering if there's a chance this will be in a different place / be renamed / etc. if we upgrade the OS at some point.

So long as we have handling for some default value, this shouldn't be an issue – I was just curious if there were corner cases we should be aware of.

Reply:
That path gets mounted here -- https://github.com/pantheon-systems/cos-daemonsets/blob/09a3a9d5d6c18c645264f3750f545bb73beda527/pauditd/devops/k8s/daemonset.yaml#L66

That shouldn't change, its what we provide to the volume mount. If the file does change, we'll change the hostMount in the daemonset yaml.

Reply:
Oh even better!  Thanks so much for clarifying, this addresses my question 😄